apps/bluesky-job-poster/src/index.ts

```diff
@@ -0,0 +1,242 @@
+import dotenv from 'dotenv'
+import { AtpAgent, RichText } from '@atproto/api'
+import { getSupabaseClient } from 'db'
+
+dotenv.config()
+
+interface Role {
+  id: string
+  title: string
+  description: string
+  currency: string
+  salary: string | null
+  company: string
+  country: string
+  language: string
+  url: string
+}
+
+interface Salary {
+  min: number
+  max: number
+  currency: string
+  period: string
+}
+
+interface Post {
+  id: string
+  title: string
+  salary: Salary
+  company: string
+  country: string
+  language: string
+  url: string
+}
+
+async function createPost() {
+  const supabaseClient = getSupabaseClient()
+  const agent = new AtpAgent({ service: 'https://bsky.social' })
+
+  try {
+    await loginToBluesky(agent)
+    const roles = await fetchRoles(supabaseClient)
+    const posts = processRoles(roles)
+    for (const post of posts) {
+      try {
+        await createSinglePost(agent, post)
+        console.log(
+          `Post created successfully for job: ${truncateString(post.title, 30)}`
+        )
+      } catch (error) {
+        console.error(
+          `Error creating post for job ${truncateString(post.title, 30)}:`,
+          error
+        )
+      }
+    }
```

Comment:
Add a delay so it doesn't spam. 

Add logs with `completed/total` with timestamp for each post.

It should take 10 minutes to post all roles, but it should send each row in the same difference of time.

For example, if there's:
- 2 roles -> each should be posted with 10 min of difference.
- 3 roles -> 1 at the start, 1 5 min after, 1 at the end.
- 4 roles -> 1 at the start, 1 2.5 min after, 1 2.5 min after, 1 at the end